### PR TITLE
Add ansible-lint to Travis

### PR DESCRIPTION
**Pull request by ehelms**
Fixes GH-1

### CODE REVIEW

**Comment by ehelms** [comment]
Figured I'd open this first in case anyone had comments on rules to omit or add before fixing them all.

File: .travis.yml [hunk 1]
```diff
@@ -2,5 +2,9 @@ sudo: false
 language: python
 python:
   - "2.7"
-install: pip install tox-travis
-script: tox
+install:
+  - pip install tox-travis
+  - pip install ansible-lint
```
Comment (evgeni) [inline .travis.yml]: time to move that to a `requirements-test.txt`?

File: rules/TrailingWhitespaceRule.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
+# Copyright (c) 2013-2014 Will Thames <will@thames.id.au>
```
Comment (evgeni) [inline rules/TrailingWhitespaceRule.py]: that is present in [ansible-lint](https://github.com/willthames/ansible-lint/blob/master/lib/ansiblelint/rules/TrailingWhitespaceRule.py) anyways, is there a reason to include it here?

**Comment by evgeni** [comment]
I wonder if we should run this via `tox` (like https://github.com/PyCQA/infrastructure/blob/master/tox.ini), and thus also running it on the developer machine when they type `tox`

File: rules/TaskVariableHasSpace.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    id = 'E305'
+    shortdesc = 'Variables should be enclosed by spaces "{{ foo }}"'
+    description = ''
+    tags = ['task']
+
+    compiled = re.compile('{{(\w*)}}')
+
+    def match(self, file, text):
+        m = self.compiled.search(text)
+        if m:
```
[the remark this answers is not in the document]
Comment (ehelms) [reply rules/TaskVariableHasSpace.py]: So `bool()` will set to True if `m` is an object class and False if `m` is `None` ?

**Comment by ehelms** [comment]
@evgeni I updated to add it to tox and modify the `.travis.yml` file. Not sure I entirely did it correct though. Is the `requirements-test.txt` needed if tox is handling it?

File: rules/TaskVariableHasSpace.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (ekohl) [reply rules/TaskVariableHasSpace.py]: Yes, ```bool()``` converts true-ish and false-ish to actual booleans:
```
>>> import re
>>> bool(re.search('{{(\w*)}}', '{{foo}}'))
True
>>> bool(re.search('{{(\w*)}}', '{{ foo }}'))
False
```

File: obal/data/tasks/brew_wait_and_download.yml [hunk 4]
```diff
@@ -1,10 +1,13 @@
 ---
 # we use this to ensure that `wait` tagged tasks only run with `--tags
 # wait` is passed
-- command: /bin/true
+- name: 'Ensure wait tags run'
+  command: /bin/true
   register: normal_execution_check
+  when: True
```
Comment (ekohl) [inline obal/data/tasks/brew_wait_and_download.yml]: What does this add?

Comment (ehelms) [reply obal/data/tasks/brew_wait_and_download.yml]: There is a check to ensure that tasks are only run when there is a change. Which means either skipping the check here or adding this always true condition.

File: tox.ini [hunk 5]
```diff
@@ -26,3 +26,10 @@ passenv = TRAVIS TRAVIS_*
 [flake8]
 exclude = .tox,*.egg,build,data
 select = E,W,F
+
+[testenv:ansible-lint]
+deps =
+    ansible
+    ansible-lint
+commands =
+    ansible-lint -R -r rules/ {toxinidir}obal/data/*.*ml
```
Comment (evgeni) [inline tox.ini]: `tox` does not run this though a shell, so you can't use wildcards :(

**Comment by evgeni** [comment]
so this works as intended for me with the following patch applied:

```diff
diff --git a/MANIFEST.in b/MANIFEST.in
index a7da3fc..2373059 100644
--- a/MANIFEST.in
+++ b/MANIFEST.in
@@ -1,2 +1,3 @@
-include requirements.txt
+include requirements*.txt
 include LICENSE
+recursive-include rules *.py
diff --git a/rules/TaskManyArgs.py b/rules/TaskManyArgs.py
index f987e57..e0c5efa 100644
--- a/rules/TaskManyArgs.py
+++ b/rules/TaskManyArgs.py
@@ -1,5 +1,6 @@
 from ansiblelint import AnsibleLintRule
 
+
 class TaskManyArgs(AnsibleLintRule):
     id = 'E303'
     shortdesc = 'Use ":" YAML format when arguments are over 3'
diff --git a/rules/TaskVariableHasSpace.py b/rules/TaskVariableHasSpace.py
index 0d496f8..2299b44 100644
--- a/rules/TaskVariableHasSpace.py
+++ b/rules/TaskVariableHasSpace.py
@@ -2,6 +2,7 @@ from ansiblelint import AnsibleLintRule
 
 import re
 
+
 class TaskVariableHasSpace(AnsibleLintRule):
     id = 'E305'
     shortdesc = 'Variables should be enclosed by spaces "{{ foo }}"'
diff --git a/tox.ini b/tox.ini
index 518a75c..7ea272a 100644
--- a/tox.ini
+++ b/tox.ini
@@ -4,7 +4,7 @@
 # and then run "tox" from this directory.
 
 [tox]
-envlist = py27
+envlist = py27,ansible-lint
 
 [testenv]
 deps =
@@ -20,7 +20,7 @@ commands =
     python setup.py check -m -s
     flake8 .
     pytest --cov={envsitepackagesdir}/obal {posargs}
-    coveralls
+    -coveralls
 passenv = TRAVIS TRAVIS_*
 
 [flake8]
@@ -32,4 +32,8 @@ deps =
     ansible
     ansible-lint
 commands =
-    ansible-lint -R -r rules/ {toxinidir}obal/data/*.*ml
+    ansible-lint -v -R -r rules/ {toxinidir}/obal/data/add_package.yml
+    ansible-lint -v -R -r rules/ {toxinidir}/obal/data/check_package.yml
+    ansible-lint -v -R -r rules/ {toxinidir}/obal/data/release_package.yml
+    ansible-lint -v -R -r rules/ {toxinidir}/obal/data/scratch_build.yml
+    ansible-lint -v -R -r rules/ {toxinidir}/obal/data/update_package.yml
```

**Comment by ekohl** [comment]
Can't you just point ansbile-lint do a directory and let it figure it out?

**Comment by evgeni** [comment]
Nope, `ansible-lint` expects either a playbook file or a role directory as param.

**Comment by ekohl** [comment]
How about ```find {toxinidir}/obal/data -name '*yaml' -or -name '*.yml' -exec ansible-lint -v -R -r rules/ {} \;```

**Comment by ehelms** [comment]
@evgeni Looks like it ran, but linting didn't run from looking at Travis output? Did i miss something?

**Comment by ekohl** [comment]
I'm guessing that's because of tox-travis interaction.

**Comment by ehelms** [comment]
Woot! This is passing now and running the linting with a wildcard. I cheated about with some stackoverflow help.

**Comment by evgeni** [comment]
Hah, nice. Didn't think of that yesterday.

I've added a `[travis]` section to `tox.ini`, so that Travis actually runs the `ansible-lint` target, and this is now IMHO good to go.

@ekohl care to ACK, as you don't have any commits in the branch? :)

File: obal/data/tasks/download_and_extract_source.yml [hunk 6]
```diff
@@ -120,7 +124,6 @@
           {{ downstream_changelog }}
         insertafter: '^%changelog'
 
-    - include: tasks/setup_sources.yml
```
Comment (ekohl) [inline obal/data/tasks/download_and_extract_source.yml]: This was no longer needed?

Comment (evgeni) [reply obal/data/tasks/download_and_extract_source.yml]: Uh. Good catch. That file does not exist anymore. So it's correct to remove the line here. But that means `update_package.yml` is broken for downstream builds (and also was already before this PR).

Comment (evgeni) [reply obal/data/tasks/download_and_extract_source.yml]: opened https://github.com/evgeni/obal/issues/24 to track this

**Pull request merged**